web/src/signalsHandlers.js

```diff
@@ -125,54 +140,59 @@ export default (store) => (signal) => {
-      // for the Profiles Cell here, not the Projects CellId which it currently has access to. 
-      // This was the source of a breaking bug
-      // if (!stateCheck.agents[payload.data.headerHash]) {
-      //   store.dispatch(
-      //     fetchAgents.create({
-      //       cellIdString: cellIdToString(cellId),
-      //       payload: null,
-      //     })
-      //   )
-      // }
-      // this one is different than the rest on purpose
-      // there's no "local action" equivalent
-      store.dispatch(setMember(cellIdToString(cellId), payload.data))
-      break
-    case SignalType.GoalWithEdge:
-      store.dispatch(
-        createSignalAction(goalActions.createGoalWithEdge, cellId, payload.data)
-      )
-      break
-    case SignalType.ArchiveGoalFully:
-      store.dispatch(
-        createSignalAction(goalActions.archiveGoalFully, cellId, payload.data)
-      )
-      break
-    default:
-      console.log('unrecognised entryType received: ', payload.entryType)
+  else {
+    // otherwise use non-crud actions
+    switch (payload.data.entryType) {
+      /*
+        PROFILES Zome
+      */
+      case SignalType.Agent:
+        // this one is different than the rest on purpose
+        // there's no "local action" equivalent
+        store.dispatch(setAgent(payload.data.data))
+        break
+      /* 
+        PROJECTS Zomes
+      */
+      case SignalType.Member:
+        const stateCheck = store.getState()
+        // check if this member is in our list of agents whose
+        // profiles we already have, if not, then we should
+        // refetch the agents list
+        // TODO: re-enable this when there's a straightforward way to have the CellId
+        // for the Profiles Cell here, not the Projects CellId which it currently has access to. 
+        // This was the source of a breaking bug
+        // if (!stateCheck.agents[payload.data.headerHash]) {
+        //   store.dispatch(
+        //     fetchAgents.create({
+        //       cellIdString: cellIdToString(cellId),
+        //       payload: null,
+        //     })
+        //   )
+        // }
+        // this one is different than the rest on purpose
+        // there's no "local action" equivalent
+        store.dispatch(setMember(cellIdToString(cellId), payload.data.data))
+        break
+      case SignalType.GoalWithEdge:
+        store.dispatch(
+          createSignalAction(goalActions.createGoalWithEdge, cellId, payload.data.data)
+        )
+        break
+      case SignalType.ArchiveGoalFully:
+        store.dispatch(
+          createSignalAction(goalActions.archiveGoalFully, cellId, payload.data.data)
+        )
+        break
+      default:
+        console.log('unrecognised entryType received: ', payload.data.entryType)
+    }
   }
 }
+function triggerRealtimeInfoAction(store, payload) {
+  // may want to check if active project is empty (or whichever way we indicate the signal 
```

Comment:
Yeah I agree, and the only data you'd need to pass along through the action would be the `agent_pub_key`